### PR TITLE
Added Basket

### PR DESCRIPTION
**Pull request by Watermelanie**
- added a baskets page
- can create, edit, delete baskets
- no synchronizations between baskets and other concepts yet

### CODE REVIEW

File: client/components/Basket/BasketComponent.vue [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  props: {
+    // Data from the stored item
+    basket: {
+      type: Object,
+      required: true
+    }
+  },
+  data() {
+    return {
+      editing: false, // Whether or not this freet is in edit mode
```
Comment (irenehuangch) [inline client/components/Basket/BasketComponent.vue]: clean up some of the comments/naming in these sections to pertain to baskets

File: server/basket/router.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ * @param {Array<{name: string, quantity: number, unit: number}>} ingredients - The items of the basket
+ * @return {BasketResponse} - The created basket
+ * @throws {403} - If the user is not logged in
+ * @throws {400} - If the item name is empty or a stream of empty spaces
+ */
+router.post(
+  '/',
+  [
+    userValidator.isUserLoggedIn,
+    basketValidator.isValidName
```
Comment (irenehuangch) [inline server/basket/router.ts:48]: should we add another middleware function or update the isValidName method to check that the new name is not already taken (no duplicates)?

File: server/basket/router.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+);
+
+/**
+ * Modify a basket's information
+ *
+ * @name PATCH /api/baskets/:basketId
+ *
+ * @param {string} name - The given name for the item
+ * @param {Array<{name: string, quantity: number, unit: number}>} ingredients - The items of the basket
+ * @return {BasketResponse} - the updated grocery item
```
Comment (irenehuangch) [inline server/basket/router.ts]: minor comment updates for basket

File: server/basket/router.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+ * @throws {404} - If the groceryItemId is not valid
+ * @throws {400} - If the item name is empty or a stream of empty spaces
+ */
+router.patch(
+  '/:basketId?',
+  [
+    userValidator.isUserLoggedIn,
+    basketValidator.isItemExists,
+    basketValidator.isValidName
+  ],
```
Comment (irenehuangch) [inline server/basket/router.ts:110]: should we add a middleware function which checks that no duplicate ingredient (ingredients with the same name perhaps?) are added?

File: client/components/Basket/BasketComponent.vue [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        Delete
+      </button>
+      <div>
+        <b>Name:</b> {{ basket.name }} <br>
+        <b>Items:</b> 
+        <li 
+          v-for="ingredient in basket.ingredients"
+          :key="ingredient.id"
+        >
+          {{ ingredient.name }} {{ ingredient.quantity }} {{ ingredient.unit }}
```
Comment (irenehuangch) [inline client/components/Basket/BasketComponent.vue]: display as "{{ item.name }} × {{ item.quantity }} {{ item.unit }}"

**Review by irenehuangch — CHANGES_REQUESTED** [review]
* add updated basket routes to readme
* merge recent commits to main branch
* some suggestions to add middleware checking for duplicate basket names and items
* render ingredients with multiplication sign
* update comments to pertain to baskets

File: server/basket/router.ts (hunk 2 above)
Comment (Watermelanie) [reply server/basket/router.ts:48]: updated isValidName to check for duplicates

File: server/basket/router.ts (hunk 4 above)
Comment (Watermelanie) [reply server/basket/router.ts:110]: i think frontend takes care of that

**Pull request merged**